microhapulator/data/template.html

```diff
@@ -338,29 +330,29 @@ <h2>Read Mapping</h2>
                         </tr>
                         <tr>
                             <th>Marker</th>
-                            {% for i in range(len(mapping_rates)) %}
+                            {% for sample in samples %}
                             <th class="alnrt" style="font-weight:bold" data-sortas="numeric">Reads</th>
                             <th class="alnrt" style="font-weight:bold" data-sortas="numeric">Repetitive</th>
                             {% endfor %}
                             <td></td>
                         </tr>
                     </thead>
                     <tbody>
-                        {% for markername in markernames %}
+                        {% for marker, marker_data in repetitive_reads_by_marker.items() %}
                         <tr>
-                            <td><a href="marker-detail-report.html?marker={{markername}}" target="_blank">{{markername}}</a></td>
-                            {% for sample_data in mapping_rates.values()%}
-                            <td class="alnrt" data-sortvalue="{{sample_data.loc[markername, 'ReadCount']}}">{{"{:,d}".format(sample_data.loc[markername, 'ReadCount'])}}</td>
-                            <td class="alnrt" data-sortvalue="{{sample_data.loc[markername, 'RepetitiveReads']}}">{{"{:,d}".format(sample_data.loc[markername, 'RepetitiveReads'])}}</td>
+                            <td><a href="marker-detail-report.html?marker={{marker}}" target="_blank">{{marker}}</a></td>
+                            {% for sample in samples %}
+                            <td class="alnrt" data-sortvalue="{{ marker_data[sample].mapped }}">{{ "{:,d}".format(marker_data[sample].mapped) }}</td>
+                            <td class="alnrt" data-sortvalue="{{ marker_data[sample].repetitive }}">{{ "{:,d}".format(marker_data[sample].repetitive) }}</td>
                             {% endfor %}
                         </tr>
                         {% endfor %}
                     </tbody>
                 </table>
             </div>
 
-            {% for plot in plots["locbalance"] %}
-            <img src="{{plot}}" />
+            {% for sample in samples %}
+            <img src="analysis/{{sample}}/{{sample}}-interlocus-balance.png" />
```

Comment:
Storing a dictionary of plots was a vestige from when images were embedded directly as base64-encoded data rather than as separate files. It turns out that the `plots` argument isn't really necessary any more. This code is more expressive.